ledger-service/http-json/src/bench/scala/com/daml/http/dbbackend/ContractDaoBenchmark.scala

```diff
@@ -107,5 +100,87 @@ abstract class ContractDaoBenchmark extends OracleAround {
       .unsafeRunSync()
     assert(inserted == batchSize)
   }
+}
+
+trait BenchmarkDbConnection {
+  def connectToDb(): Unit
+  def createDbJdbcConfig: JdbcConfig
+  def cleanup(): Unit
+}
+
+trait OracleBenchmarkDbConn extends BenchmarkDbConnection with OracleAround {
+
+  private var user: User = _
+  private val disableContractPayloadIndexing = false
+
+  override def connectToDb() = {
+    connectToOracle()
+  }
 
+  override def createDbJdbcConfig: JdbcConfig = {
+    user = createNewRandomUser()
+    val cfg = JdbcConfig(
+      dbutils.JdbcConfig(
+        driver = "oracle.jdbc.OracleDriver",
+        url = oracleJdbcUrl,
+        user = user.name,
+        password = user.pwd,
+        poolSize = ConnectionPool.PoolSize.Integration,
+      ),
+      dbStartupMode = DbStartupMode.CreateOnly,
+      backendSpecificConf =
+        if (disableContractPayloadIndexing) Map(DisableContractPayloadIndexing -> "true")
+        else Map.empty,
+    )
+    cfg
+  }
+
+  override def cleanup(): Unit = {
+    dropUser(user.name)
+  }
+}
+
+/*
+ Running the PG benchmark requires us to explicitly set the PGPASSWORD env variable which is picked
+ by the `createdb` tool when creating relevant database e.g
+
+ $ PGPASSWORD=<password> bazel run //ledger-service/http-json:contractdao-bench -- -f 1 -i 1 -wi 1 -bm avgt \
+ "QueryPayloadBenchmarkPostgres" -p extraParties=1,10 -p extraPayloadValues=1,10,100 -p batchSize=1000
+ */
+trait PostgresBenchmarkDbConn extends BenchmarkDbConnection with PostgresAround {
+
+  @volatile
+  private var database: PostgresDatabase = _
+
+  private val host = sys.env.getOrElse("POSTGRESQL_HOST", "localhost")
+  private val port = Port(sys.env.get("POSTGRESQL_PORT").map(_.toInt).getOrElse(5432))
+  private val user = sys.env.getOrElse("POSTGRESQL_USERNAME", "postgres")
+  private val password = sys.env.getOrElse(
+    "PGPASSWORD",
+    throw new IllegalArgumentException(
+      s"Missing required PGPASSWORD env value, needed to run `createdb` "
+    ),
+  )
+
+  override def connectToDb() = {
+    connectToSharedServer(host, port, user, password)
```

Comment:
Why  not use `connectToPostgresqlServer`, which is already accessible, here? It also embeds the envvar lookups.

Reply:
Something like (combines both prior comments, I'm missing a detail somewhere though):

```diff
diff --git a/ledger-service/http-json/src/bench/scala/com/daml/http/dbbackend/ContractDaoBenchmark.scala b/ledger-service/http-json/src/bench/scala/com/daml/http/dbbackend/ContractDaoBenchmark.scala
index fe1584af5c6..56043b2af95 100644
--- a/ledger-service/http-json/src/bench/scala/com/daml/http/dbbackend/ContractDaoBenchmark.scala
+++ b/ledger-service/http-json/src/bench/scala/com/daml/http/dbbackend/ContractDaoBenchmark.scala
@@ -14,7 +14,6 @@ import com.daml.http.dbbackend.Queries.{DBContract, SurrogateTpId}
 import com.daml.http.domain.TemplateId
 import com.daml.http.util.Logging.instanceUUIDLogCtx
 import com.daml.metrics.Metrics
-import com.daml.ports.Port
 import com.daml.testing.oracle
 import com.daml.testing.postgresql.{PostgresAround, PostgresDatabase}
 import oracle.{OracleAround, User}
@@ -152,18 +151,8 @@ trait PostgresBenchmarkDbConn extends BenchmarkDbConnection with PostgresAround
   @volatile
   private var database: PostgresDatabase = _
 
-  private val host = sys.env.getOrElse("POSTGRESQL_HOST", "localhost")
-  private val port = Port(sys.env.get("POSTGRESQL_PORT").map(_.toInt).getOrElse(5432))
-  private val user = sys.env.getOrElse("POSTGRESQL_USERNAME", "postgres")
-  private val password = sys.env.getOrElse(
-    "PGPASSWORD",
-    throw new IllegalArgumentException(
-      s"Missing required PGPASSWORD env value, needed to run `createdb` "
-    ),
-  )
-
   override def connectToDb() = {
-    connectToSharedServer(host, port, user, password)
+    connectToPostgresqlServer()
     database = createNewRandomDatabase()
   }
 
@@ -172,8 +161,8 @@ trait PostgresBenchmarkDbConn extends BenchmarkDbConnection with PostgresAround
       dbutils.JdbcConfig(
         driver = "org.postgresql.Driver",
         url = database.url,
-        user = user,
-        password = password,
+        user = server.userName,
+        password = server.password,
         poolSize = ConnectionPool.PoolSize.Integration,
       ),
       dbStartupMode = DbStartupMode.CreateOnly,
@@ -182,5 +171,6 @@ trait PostgresBenchmarkDbConn extends BenchmarkDbConnection with PostgresAround
 
   override def cleanup(): Unit = {
     dropDatabase(database)
+    disconnectFromPostgresqlServer()
   }
 }
diff --git a/libs-scala/postgresql-testing/src/main/scala/com/digitalasset/testing/postgresql/PostgresAround.scala b/libs-scala/postgresql-testing/src/main/scala/com/digitalasset/testing/postgresql/PostgresAround.scala
index abe30b344c4..9bd39b10268 100644
--- a/libs-scala/postgresql-testing/src/main/scala/com/digitalasset/testing/postgresql/PostgresAround.scala
+++ b/libs-scala/postgresql-testing/src/main/scala/com/digitalasset/testing/postgresql/PostgresAround.scala
@@ -20,7 +20,8 @@ import scala.util.Try
 import scala.util.control.NonFatal
 
 trait PostgresAround {
-  @volatile private var server: PostgresServer = _
+  protected[this] def server = server_
+  @volatile private var server_ : PostgresServer = _
   @volatile private var paths: Option[PostgresServerPaths] = None
 
   private val started: AtomicBoolean = new AtomicBoolean(false)
@@ -46,7 +47,7 @@ trait PostgresAround {
       password: String,
   ): Unit = {
     logger.info(s"Connected to PostgreSQL on $hostName:$port.")
-    server = PostgresServer(hostName, port, userName, password)
+    server_ = PostgresServer(hostName, port, userName, password)
   }
 
   private def startEphemeralServer(): Unit = {
@@ -60,7 +61,7 @@ trait PostgresAround {
     val port = lockedPort.port
     val userName = "test"
     val password = ""
-    server = PostgresServer(hostName, port, userName, password)
+    server_ = PostgresServer(hostName, port, userName, password)
     paths = Some(PostgresServerPaths(root, dataDir, logFile))
 
     try {
@@ -85,7 +86,7 @@ trait PostgresAround {
       deleteRecursively(root)
       logger.info("PostgreSQL has stopped, and the data directory has been deleted.")
     }
-    server = null
+    server_ = null
   }
 
   private def startPostgresql(dataDir: Path, logFile: Path): Unit = {
```

Reply:
made the changes 👍